.github/workflows/main.yml

```
@@ -0,0 +1,41 @@
+name: Build and test
+
+on:
+  push:
+  pull_request:
+  merge_group:
+
+concurrency: # On new push, cancel old workflows from the same PR, branch or tag:
+  group: ${{ github.workflow }}-${{github.event_name}}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies (apt)
+        run: sudo apt-get install -y libxml2-dev libxen-dev
+
```

The updates may fail if the repositories are not up-to-date:
```suggestion
      - name: Update Ubuntu repositories
        run: sudo apt-get update

      - name: Install dependencies (apt)
        run: sudo apt-get install -y libxml2-dev libxen-dev
```